src/prof_data.c

```diff
@@ -56,6 +56,20 @@ static ckh_t bt2gctx;
  */
 static prof_tdata_tree_t tdatas;
 
+/* The following are needed for dumping and are protected by prof_dump_mtx. */
```

Comment:
The intended end state for this code is for this code is to rewrite it in terms of (e.g.) the buffered writer API, right?

Rather than passing these as globals, why not pull them into a `prof_output_ctx_t` or something that contains them as well as `prof_dump_fd`, and we can just propagate a pointer to that context struct everywhere. On its own, this change is sort of a sideways move; it highlights the non-error case better, but at the cost of making the flow of data much harder to reason about.

Reply:
> The intended end state for this code is for this code is to rewrite it in terms of (e.g.) the buffered writer API, right?

Yes.

> Rather than passing these as globals, why not pull them into a `prof_output_ctx_t` or something that contains them as well as `prof_dump_fd`, and we can just propagate a pointer to that context struct everywhere. On its own, this change is sort of a sideways move; it highlights the non-error case better, but at the cost of making the flow of data much harder to reason about.

Good point. I'll have that in the restructuring commits. For now the commit is more for the functionality change, and I'll put up commits for restructuring later (which will keep the functionality exactly the same and all the tests passing), but thanks for the early feedback.